tests/backup/backup.go

```diff
@@ -0,0 +1,567 @@
+	AsyncCommit bool
+	OnePC       bool
+}
+
+// Config means the config of this test case
+type Config struct {
+	NumAccounts int
+	Concurrency int
+	Contention  string
+	// run backup once every BackupInterval
+	BackupInterval time.Duration
+	// run restore once every RestoreInterval
+	RestoreInterval time.Duration
+	DbName          string
+	RetryLimit      int
+	// will backup to BackupURI/full-$nextBackupIndex
+	BackupURI string
+}
+
+type backupClient struct {
+	features         Features
+	config           Config
+	db               *sql.DB
+	txnID            int32
+	lastBackupTs     uint64
+	nextRestoreIndex int
+	nextBackupIndex  int
+}
+
+func randomString(n int) string {
+	var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+
+	s := make([]rune, n)
+	for i := range s {
+		s[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(s)
+}
+
+func (c *backupClient) applyConfig() {
+	var err error
+	stmt := fmt.Sprintf("set @@tidb_replica_read = '%s'", c.features.ReplicaRead)
+	if _, err = c.db.Exec(stmt); err != nil {
+		log.Errorf("[%s] tidb_replica_read set fail: %v", c, err)
+	}
+	if c.features.AsyncCommit {
+		_, err = c.db.Exec("set @@global.tidb_enable_async_commit = 1;")
+	} else {
+		_, err = c.db.Exec("set @@global.tidb_enable_async_commit = 0;")
+	}
+	if err != nil {
+		log.Fatalf("[%s] set async commit failed: %v", c, err)
+	}
+	if c.features.OnePC {
+		_, err = c.db.Exec("set @@global.tidb_enable_1pc = 1;")
+	} else {
+		_, err = c.db.Exec("set @@global.tidb_enable_1pc = 0;")
+	}
+	if err != nil {
+		log.Fatalf("[%s] set 1PC failed: %v", c, err)
+	}
+	if c.features.Pessimistic {
+		_, err = c.db.Exec("set @@global.tidb_txn_mode = 'pessimistic';")
+	} else {
+		_, err = c.db.Exec("set @@global.tidb_txn_mode = 'optimistic';")
+	}
+	if err != nil {
+		log.Fatalf("[%s] set txn_mode failed: %v", c, err)
+	}
+	time.Sleep(5 * time.Second)
+}
+
+func (c *backupClient) createTables() {
+	for _, stmt := range stmtsCreate {
+		if _, err := c.db.Exec(stmt); err != nil {
+			log.Fatalf("[%s] execute statement %s error %v", c, stmt, err)
+		}
+	}
+}
+
+func (c *backupClient) initData(ctx context.Context) {
+	var wg sync.WaitGroup
+	for i := 0; i < c.config.NumAccounts; i++ {
+		stmt := fmt.Sprintf(`INSERT IGNORE INTO accounts (id, balance, name, remark) VALUES (%d, %d, "account %d", "%s");`, i, initialBalance, i, randomString(36))
+		wg.Add(1)
+		go func(db *sql.DB) {
+			defer wg.Done()
+			err := util.RunWithRetry(ctx, c.config.RetryLimit, 5*time.Second, func() error {
+				_, err := db.Exec(stmt)
+				if util.IsErrDupEntry(err) {
+					return nil
+				}
+				return err
+			})
+			if err != nil {
+				log.Fatalf("[%s] exec %s err %v", c, stmt, err)
+			}
+		}(c.db)
+	}
+	wg.Wait()
+}
+
+func (c *backupClient) backup() error {
+	log.Infof("[%s] Try backup once", c)
+	queryString := fmt.Sprintf(`BACKUP DATABASE * TO '%s/full-%d' LAST_BACKUP = %d;`, c.config.BackupURI, c.nextBackupIndex, c.lastBackupTs)
```

Comment:
On K8s env, we want to set `BackupURI` to `s3://bucket/path/to?key=xx&secret=yy&endpoint=zz&opt=value`.

Maybe we should use `net/url` to Encode the url so that it is compatible with both `BackupURI="s3://xxx/yyy?zz=1"` and `BackupURI="local:///tmp/backup"`.